### DISCUSSION TITLE
Does java agent support spring scheduler and spring event listeners?

### DISCUSSION THREAD

**Discussion by johnvlittle** [open-telemetry/opentelemetry-java-instrumentation#12388] [Q&A]
We are using spring boot 2.0.3. we see the agent is sending metrics and traces for REST API calls, but some of our services have no rest calls, code is triggered by either listening for an event or a scheduled event.  Are either of these picked up by the java auto agent?  We cant find them in New relic, but that doesnt mean they are not there.

If we need to manually instrument scheduled job code and event listener code, we assume we cant use the open telemetry spring boot library as it doesnt support 2.0.3 (only 2.6+). In this case, we assume our only option would be micrometer?

**Comment by trask** [comment]
Spring Scheduling is supported, for full list check out: https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/docs/supported-libraries.md

You could test by using the console exporter to see what is captured in case that helps.
